### PR TITLE
feat: remove hardhat from relayer code

### PR DESCRIPTION
**Pull request by cehali**

### CODE REVIEW

**Review by koonopek — APPROVED** [review]
Overall looks good! 
I'm only wondering if we could tune typescript to work with hardhat, but you have probably tried :)

File: packages/on-chain-relayer/test/relayer/update-prices.test.ts [hunk 1]
```diff
@@ -58,6 +59,8 @@ describe("#updatePrices", () => {
   });
 
   it("should update prices in mento adapter", async () => {
+    (getProviderOrSigner as any).getProvider = () => ethers.provider;
```
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/on-chain-relayer/test/relayer/update-prices.test.ts:62]: I see I will have to add in my PR anyway

**Review by hatskier — COMMENTED** [review]
Please wait before merging this

File: packages/on-chain-relayer/scripts/mento/deploy-mock-sorted-oracles.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { Signer } from "ethers";
+import { ethers } from "hardhat";
+
+export const deployMockSortedOracles = async (signer?: Signer) => {
```
Comment (hatskier) [inline packages/on-chain-relayer/scripts/mento/deploy-mock-sorted-oracles.ts:4]: IMO this is not a script and should not be located in the scripts folder. Script should be executable, but this file just exports a function

File: packages/on-chain-relayer/src/custom-integrations/mento/mento-utils.ts [hunk 3]
```diff
@@ -95,29 +94,3 @@ export const prepareLinkedListLocationsForMentoAdapterReport = async ({
 
   return locationsInSortedLinkedLists;
 };
-
-export const deployMockSortedOracles = async (signer?: Signer) => {
```
Comment (hatskier) [inline packages/on-chain-relayer/src/custom-integrations/mento/mento-utils.ts:99]: IMO this is a good location for this function

**Review by hatskier — CHANGES_REQUESTED** [review]

File: packages/on-chain-relayer/src/custom-integrations/mento/mento-utils.ts (hunk 3 above)
Comment (cehali) [reply packages/on-chain-relayer/src/custom-integrations/mento/mento-utils.ts:99]: But it is not used anywhere in files from the src folder. It is only used in one script and tests.

**Pull request merged**